src/OpenTelemetry/Collector/DiagnosticSourceSubscriber.cs

```diff
@@ -71,8 +86,13 @@ public void Dispose()
                 return;
             }
 
-            this.listenerSubscription?.Dispose();
-            this.listenerSubscription = null;
+            lock (this.listenerSubscriptions)
+            {
+                foreach (var listenerSubscription in this.listenerSubscriptions)
+                {
+                    listenerSubscription?.Dispose();
+                }
```

Comment:
Would it make sense to clear the list here to drop the references?
Something like:
```suggestion
                }
                this.listenerSubscriptions.Clear();
```

Reply:
I don't think it really matters but I can do the change.